src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,277 @@
+                        // fuse existing convert
+                        auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                            existing_convert->get_input_node_shared_ptr(0),
+                            required_after);
+                        ngraph::copy_runtime_info(parent_output.get_node_shared_ptr(), convert);
+                        op->set_argument(op_input.get_index(), convert);
+                        continue;
+                    }
+
+                    // create new convert
+                    auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                        existing_convert->output(0),
+                        required_after);
+                    ngraph::copy_runtime_info(existing_convert->output(0).get_node()->shared_from_this(), convert);
+                    op->set_argument(op_input.get_index(), convert);
+                }
+            }
+        }
+
+        auto type_relaxed_node = std::dynamic_pointer_cast<ov::op::TypeRelaxedBase>(op);
+        if (input_precisions_were_changed || (type_relaxed_node != nullptr)) {
+            // update output precision
+            std::vector<element::Type> op_output_types;
+            for (auto& output : op->outputs()) {
+                op_output_types.push_back(output.get_element_type());
+            }
+
+            if (type_relaxed_node != nullptr) {
+                // TODO: user story 104281
+                // to keep previous functionality
+                // unary and binary element-wise operations are supported
+                // will be replaced to snippets opset later
+                const auto op_element_type = op->get_input_source_output(0).get_element_type();
+                if (type_relaxed_node->get_overridden_output_type(0) != op_element_type) {
+                    assert(op->get_output_size() == 1ull);
+
+                    type_relaxed_node->set_overridden_output_type(op_element_type, 0);
+                    std::dynamic_pointer_cast<ngraph::Node>(op)->validate_and_infer_types();
+                }
+            } else {
+                op->validate_and_infer_types();
+            }
+
+            for (auto i = 0ull; i < op->get_output_size(); ++i) {
+                const auto& output = op->output(i);
+                if (output.get_element_type() != op_output_types[i]) {
+                    auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                        output,
+                        op_output_types[i]);
+                    ngraph::copy_runtime_info(output.get_node_shared_ptr(), convert);
+                    for (auto& input : output.get_target_inputs()) {
+                        auto child = input.get_node();
+                        if (child == convert.get()) {
+                            continue;
+                        }
+                        input.replace_source_output(convert->output(0));
```

Comment:
If it is applicable I suggest using `ov::replace_output_update_name` here

Reply:
`convert` is created as consumer for `output`. In this case `replace_output_update_name` works incorrectly. My suggestion is: keep it as is.

Reply:
Thanks for the explanation, I agree